app/js/lib/foxbox/settings.js

```diff
@@ -29,29 +29,26 @@ const storage = localStorage ? localStorage : {
   clear: () => {},
 };
 
-export default class Settings extends Model {
+export default class Settings extends EventDispatcher {
   constructor() {
+    super(['session-changed', 'polling-setting-changed']);
+
     const localOrigin = storage.getItem(`${PREFIX}localOrigin`);
 
     const pollingEnabled = storage.getItem(`${PREFIX}pollingEnabled`) !== null ?
       storage.getItem(`${PREFIX}pollingEnabled`) === 'true' :
       DEFAULT_POLLING_ENABLED;
 
-    super({
-      _configured: storage.getItem(`${PREFIX}configured`) !== null ?
-      storage.getItem(`${PREFIX}configured`) === 'true' : false,
-
-      _localOrigin: localOrigin,
-      _tunnelOrigin: storage.getItem(`${PREFIX}tunnelOrigin`) || '',
-      _clientId: storage.getItem(`${PREFIX}clientId`) || '',
-
-      _session: storage.getItem(`${PREFIX}session`),
-      _skipDiscovery: storage.getItem(`${PREFIX}skipDiscovery`) === 'true',
-      _pollingEnabled: pollingEnabled,
-
-      _pushEndpoint: storage.getItem(`${PREFIX}push_endpoint`) || null,
-      _pushPubKey: storage.getItem(`${PREFIX}push_pubKey`) || null,
-    });
+    this._configured = storage.getItem(`${PREFIX}configured`) !== null ?
+      storage.getItem(`${PREFIX}configured`) === 'true' : false;
+    this._localOrigin = localOrigin;
+    this._tunnelOrigin = storage.getItem(`${PREFIX}tunnelOrigin`) || '';
+    this._clientId = storage.getItem(`${PREFIX}clientId`) || '';
+    this._session = storage.getItem(`${PREFIX}session`);
+    this._skipDiscovery = storage.getItem(`${PREFIX}skipDiscovery`) === 'true';
+    this._pollingEnabled = pollingEnabled;
+    this._pushEndpoint = storage.getItem(`${PREFIX}push_endpoint`) || null;
+    this._pushPubKey = storage.getItem(`${PREFIX}push_pubKey`) || null;
```

Comment:
These all should really be private members exposed via getters and optional setters.
If not in this PR, we should create an issue to keep a track of it.